mne/io/nsx/nsx.py

```diff
@@ -0,0 +1,539 @@
+                    'mV': 1e-3
+                    }
+
+nsx_header_dict = {
+    'basic_2.1': [
+        ('file_id', 'S8'),
+        # label of sampling group (e.g. "1kS/s" or "LFP Low")
+        ('label', 'S16'),
+        # number of 1/30000 seconds between data points
+        # (e.g., if sampling rate "1 kS/s", period equals "30")
+        ('period', 'uint32'),
+        ('channel_count', 'uint32')],
+    'basic': [
+        ('file_id', 'S8'),  # achFileType
+        # file specification split into major and minor version number
+        ('ver_major', 'uint8'),
+        ('ver_minor', 'uint8'),
+        # bytes of basic & extended header
+        ('bytes_in_headers', 'uint32'),
+        # label of the sampling group (e.g., "1 kS/s" or "LFP low")
+        ('label', 'S16'),
+        ('comment', 'S256'),
+        ('period', 'uint32'),
+        ('timestamp_resolution', 'uint32'),
+        # time origin: 2byte uint16 values for ...
+        ('year', 'uint16'),
+        ('month', 'uint16'),
+        ('weekday', 'uint16'),
+        ('day', 'uint16'),
+        ('hour', 'uint16'),
+        ('minute', 'uint16'),
+        ('second', 'uint16'),
+        ('millisecond', 'uint16'),
+        # number of channel_count match number of extended headers
+        ('channel_count', 'uint32')],
+    'extended': [
+        ('type', 'S2'),
+        ('electrode_id', 'uint16'),
+        ('electrode_label', 'S16'),
+        # used front-end amplifier bank (e.g., A, B, C, D)
+        ('physical_connector', 'uint8'),
+        # used connector pin (e.g., 1-37 on bank A, B, C or D)
+        ('connector_pin', 'uint8'),
+        # digital and analog value ranges of the signal
+        ('min_digital_val', 'int16'),
+        ('max_digital_val', 'int16'),
+        ('min_analog_val', 'int16'),
+        ('max_analog_val', 'int16'),
+        # units of the analog range values ("mV" or "uV")
+        ('units', 'S16'),
+        # filter settings used to create nsx from source signal
+        ('hi_freq_corner', 'uint32'),
+        ('hi_freq_order', 'uint32'),
+        ('hi_freq_type', 'uint16'),  # 0=None, 1=Butterworth
+        ('lo_freq_corner', 'uint32'),
+        ('lo_freq_order', 'uint32'),
+        ('lo_freq_type', 'uint16')],  # 0=None, 1=Butterworth,
+    'data>2.1<3': [
+        ('header', 'uint8'),
+        ('timestamp', 'uint32'),
+        ('nb_data_points', 'uint32')],
+    'data>=3': [
+        ('header', 'uint8'),
+        ('timestamp', 'uint64'),
+        ('nb_data_points', 'uint32')],
+}
+
+
+def read_raw_nsx(nsx_fname, stim_channel=True, eog=[], misc=[], preload=False):
+    return RawNSX(nsx_fname, stim_channel, eog, misc, preload=preload)
+
+
+class RawNSX(BaseRaw):
+    """Raw object from NSx file from Blackrock Microsystems.
+
+    Parameters
+    ----------
+    input_fname : str
+        Path to the NSx file.
+    """
+
+    def __init__(self, nsx_fname, stim_channel=True,
+                 eog=[], misc=[], preload=False, verbose=None):
```

Comment:
all the parameters are not documented.

Also don't set a default argument to an empty list, it can have unexpected consequences. Instead the right pattern is to set it to `None` and inside the `__init__` set it to `list()` if `None`